samples/eShopLite/ServiceDefaults/Extensions.cs

```diff
@@ -56,6 +57,12 @@ public static IHostApplicationBuilder ConfigureOpenTelemetry(this IHostApplicati
                 tracing.AddAspNetCoreInstrumentation()
                        .AddGrpcClientInstrumentation()
                        .AddHttpClientInstrumentation();
+            })
+            .ConfigureResource(rb => {
+                rb.AddAttributes(new Dictionary<string, object>
+                {
+                    ["process.id"] = Environment.ProcessId,
```

Comment:
cc @JamesNK

Reply:
The OTEL way to identify resources is with `service.name`, `service.instance.id` and `service.namespace`. We should try to use them.

Spec: https://opentelemetry.io/docs/specs/semconv/resource/

Reply:
Some thoughts here:
* `service.instance.id` is a GUID that comes from the hosting UID. It's a great unique identifier from a computer's perspective, but lousy from a human's perspective.
* Resource attributes can be configured from env vars (it's what we are doing for `service.instance.id`) but that's not possible for PID because we don't know the `process.id` at this point.
* The problem with using PID as an identifier is hosting/dashboard can't define it ahead of time. See bullet point above.

Different approach:
* An alternative to OTEL sending the PID with its telemetry is matching the OTEL resource to the Aspire resource using the UID and getting the PID from the resource.
* How well is PID going to work in cloud scenarios?